include/ddc/kernels/splines/spline_evaluator_2d_batched.hpp

```diff
@@ -0,0 +1,508 @@
+                KOKKOS_CLASS_LAMBDA(typename batch_domain_type::discrete_element_type const j) {
+                    const auto spline_eval_2D = spline_eval[j];
+                    const auto coords_eval_2D = coords_eval[j];
+                    const auto spline_coef_2D = spline_coef[j];
+                    for (auto const i1 : interpolation_domain1) {
+                        for (auto const i2 : interpolation_domain2) {
+                            spline_eval_2D(i1, i2) = eval(coords_eval_2D(i1, i2), spline_coef_2D);
+                        }
+                    }
+                });
+    }
+
+    template <class Layout, class... CoordsDims>
+    double deriv_dim_1(
+            ddc::Coordinate<CoordsDims...> const& coord_eval,
+            ddc::ChunkSpan<double const, bsplines_domain_type, Layout, memory_space> const
+                    spline_coef) const
+    {
+        return eval_no_bc<eval_deriv_type, eval_type>(coord_eval, spline_coef);
+    }
+
+    template <class Layout, class... CoordsDims>
+    double deriv_dim_2(
+            ddc::Coordinate<CoordsDims...> const& coord_eval,
+            ddc::ChunkSpan<double const, bsplines_domain_type, Layout, memory_space> const
+                    spline_coef) const
+    {
+        return eval_no_bc<eval_type, eval_deriv_type>(coord_eval, spline_coef);
+    }
+
+    template <class Layout, class... CoordsDims>
+    double deriv_1_and_2(
+            ddc::Coordinate<CoordsDims...> const& coord_eval,
+            ddc::ChunkSpan<double const, bsplines_domain_type, Layout, memory_space> const
+                    spline_coef) const
+    {
+        return eval_no_bc<eval_deriv_type, eval_deriv_type>(coord_eval, spline_coef);
+    }
+
+    template <class InterestDim, class Layout, class... CoordsDims>
+    double deriv(
+            ddc::Coordinate<CoordsDims...> const& coord_eval,
+            ddc::ChunkSpan<double const, bsplines_domain_type, Layout, memory_space> const
+                    spline_coef) const
+    {
+        static_assert(
+                std::is_same_v<
+                        InterestDim,
+                        typename interpolation_mesh_type1::
+                                continuous_dimension_type> || std::is_same_v<InterestDim, typename interpolation_mesh_type2::continuous_dimension_type>);
+        if constexpr (std::is_same_v<
+                              InterestDim,
+                              typename interpolation_mesh_type1::continuous_dimension_type>) {
+            return deriv_dim_1(coord_eval, spline_coef);
+        } else if constexpr (std::is_same_v<
+                                     InterestDim,
+                                     typename interpolation_mesh_type2::
+                                             continuous_dimension_type>) {
+            return deriv_dim_2(coord_eval, spline_coef);
+        }
+    }
+
+    template <class InterestDim1, class InterestDim2, class Layout, class... CoordsDims>
+    double deriv2(
```

Comment:
Would like to call it deriv() but CTAD do not differentiate if from previous function